tests/integration/calendar/test_date_picker_persistence.py

```diff
@@ -0,0 +1,158 @@
+
+    dash_dcc.clear_input(dps1)
+    send_date_dps(dps1, "01/01/2020")
+    dash_dcc.wait_for_text_to_equal("#dps1-p", "2020-01-01")
+
+    dash_dcc.clear_input(dps2)
+    send_date_dps(dps2, "01/01/2020")
+    dash_dcc.wait_for_text_to_equal("#dps2-p", "2020-01-01")
+
+    dash_dcc.find_element("#btn").click()
+
+    dash_dcc.wait_for_text_to_equal("#dps1-p", "2020-01-01")
+    dash_dcc.wait_for_text_to_equal("#dps2-p", "2020-01-01")
+
+
+def test_rdpr002_persisted_dpr(dash_dcc):
+    def send_date_dpr(target, start_date, end_date):
+        (
+            ActionChains(dash_dcc.driver)
+            .move_to_element(target)
+            .click(target)
+            .send_keys(Keys.END)
+            .key_down(Keys.SHIFT)
+            .send_keys(Keys.HOME)
+            .key_up(Keys.SHIFT)
+            .send_keys(str(start_date))
+            .pause(0.2)
+            .send_keys(Keys.END)
+            .key_down(Keys.SHIFT)
+            .send_keys(Keys.HOME)
+            .key_up(Keys.SHIFT)
+            .send_keys(str(end_date))
+        ).perform()
+
+    app = dash.Dash(__name__)
+    app.layout = html.Div(
+        [
+            html.Button("fire callback", id="btn"),
+            html.Div(
+                children=[
+                    dcc.DatePickerRange(
+                        id="dpr1",
+                        start_date=datetime.today() - timedelta(days=3),
+                        end_date=datetime.today(),
+                        persistence=True,
+                        persistence_type="session",
+                    ),
+                    html.P("dpr1", id="dpr1-p-start"),
+                    html.P("dpr1", id="dpr1-p-end"),
+                    html.Div(id="container"),
+                    html.P("dpr2", id="dpr2-p-start"),
+                    html.P("dpr2", id="dpr2-p-end"),
+                ]
+            ),
+        ]
+    )
+
+    @app.callback(Output("container", "children"), [Input("btn", "n_clicks")])
+    def update_output(value):
+        return dcc.DatePickerRange(
+            id="dpr2",
+            start_date=datetime.today() - timedelta(days=3),
+            end_date=datetime.today(),
+            persistence=True,
+            persistence_type="session",
+        )
+
+    @app.callback(Output("dpr1-p-start", "children"), [Input("dpr1", "start_date")])
+    def display_dps1(value):
+        return value
+
+    @app.callback(Output("dpr1-p-end", "children"), [Input("dpr1", "end_date")])
+    def display_dps1(value):
+        return value
+
+    @app.callback(Output("dpr2-p-start", "children"), [Input("dpr2", "start_date")])
+    def display_dps2(value):
+        return value
+
+    @app.callback(Output("dpr2-p-end", "children"), [Input("dpr2", "end_date")])
+    def display_dps2(value):
+        return value
+
+    dash_dcc.start_server(app)
+    dpr1 = dash_dcc.find_element("div#dpr1 div div div div .DateInput_input")
+    dpr2 = dash_dcc.find_element("div#dpr2 div div div div .DateInput_input")
```

Comment:
Does it work with only `div#dpr2 .DateInput_input` ? This `find_element` seems to be very dependent on the implementation details of the underlying third-party component

Reply:
See 605594c . Now we use only the `dash_dcc` functions for interacting with date pickers, and are testing the behaviour DatePickerSingle and Range defined in callbacks, with the `date`, `start/end_date` props changing value each time the callback is fired.